Modules/_testcapimodule.c

```diff
@@ -2340,6 +2340,177 @@ get_timezone_utc_capi(PyObject* self, PyObject *args) {
     }
 }
 
+static PyObject *
+get_date_fromdate(PyObject *self, PyObject *args)
+{
+    PyObject *rv = NULL;
+    int year = 0, month = 0, day = 0;
+    int macro = 0;
+
+    if (!PyArg_ParseTuple(args, "iii|p", &year, &month, &day, &macro)) {
+        return NULL;
+    }
+
+    // Pass along to the API function
+    if (macro) {
+        rv = PyDate_FromDate(year, month, day);
+    }
+    else {
+        rv = PyDateTimeAPI->Date_FromDate(
+            year, month, day,
+            PyDateTimeAPI->DateType);
+    }
+
```

Comment:
nitpick: I suggest this empty line ;-) It doesn't add any value in term of readability.

My comments apply to all C functions below.